nexus/db-queries/src/db/raw_query_builder.rs

```diff
@@ -0,0 +1,151 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Utilities for building string-based queries.
+//!
+//! These largely side-step Diesel's type system,
+//! and are recommended for more complex CTE
+
+use diesel::pg::Pg;
+use diesel::sql_types;
+use std::cell::Cell;
+
+// Keeps a counter to "how many bind parameters have been used" to
+// aid in the construction of the query string.
+struct BindParamCounter(Cell<i32>);
+impl BindParamCounter {
+    fn new() -> Self {
+        Self(0.into())
+    }
+    fn next(&self) -> i32 {
+        self.0.set(self.0.get() + 1);
+        self.0.get()
+    }
+}
+
+/// A "trusted" string, which can be used to construct SQL queries even
+/// though it isn't static. We use "trust" to refer to "protection from
+/// SQL injections".
+///
+/// This is basically a workaround for cases where we haven't yet been
+/// able to construct a query at compile-time.
+pub enum TrustedStr {
+    Static(&'static str),
+    ValidatedExplicitly(String),
+}
+
+impl TrustedStr {
+    /// Explicitly constructs a string, with a name that hopefully
+    /// gives callers some pause when calling this API.
+    ///
+    /// If arbitrary user input is provided here, this string COULD
+    /// cause SQL injection attacks, so each call-site should have a
+    /// justification for "why it's safe".
+    pub fn this_string_wont_cause_sql_injections(s: String) -> Self {
```

Comment:
“this string could” reads like more of a warning that “this string won’t”, which could be misinterpreted as a reassurance

Reply:
Updated to `i_take_responsibility_for_validating_this_string` - I'm okay wordsmithing this, since IMO it should be long, obvious, and hard to accidentally call.

Also wrapped this in a struct so a caller can't create the enum. I always forget that all enum variants are always accessible for construction.